### DISCUSSION TITLE
getting element id from wire:id

### DISCUSSION THREAD

**Discussion by hyphen81** [livewire/livewire#4785] [1. Help]
I have sort of 2 questions. I'm using a livewire full-page component to display a list of tasks along with all the CRUD stuff associated, and using dragula to allow for drag and drop.

I initially had all this running through a TaskListComponent, and was having trouble getting some of the functionality to work so I now have a child component that just handles the task itself. So there's a TaskListComponent that's the page, and and then I have a TaskComponent that's added to the page in a foreach loop. my thought is that all the task list type functionality would happen in the main component, and anything related to the specific task (show, update, delete, etc) would be in the TaskComponent. Tell me if that's how you would structure it?

My question now is, based on that structure, I have some javascript that's running on the page to listen for the dragula drop event and handle reordering/changing status of tasks and it relies on the element ID, but since I made each task a livewire component now the element id is being replaced with wire:id which is some hash value. I don't know how to access that in javascript and get the original html element ID.

```
@foreach ($tasks as $task)
    @if ($task->task_status_id === $status->id)
        <livewire:tasks.task-component wire:key="{{ $task->id }}" :task="$task" data-task-id="{{ $task->id }}" />
    @endif
@endforeach
```

```
drag.on('drop', function(el, target, source, sibling) {
                console.log(el);
});
```

The above console.log outputs:

`<div wire:id="1pPdyV6eX8kgtePfXBzk" ...></div>`

Sorry I know this is a noob question, but it's not clear from the docs how to get to what I'm after. I've tried using Livewire.find() but it returns undefined. I can use Livewire command in the console, so I know it's available.

Also possibly related, any kind of $wire command I call says $wire is undefined. I've got this in my app.js file that's loaded after livewire scripts:

```
import Alpine from 'alpinejs'
window.Alpine = Alpine
Alpine.start()
```

And I can do Alpine in the console and it knows what that is.

Any help would be greatly appreciated.

**Comment by joshhanley** [comment]
@hyphen81 you can't pass "attributes" into a Livewire component because it has props, which get passed into the `mount` method.

But you can output stuff on the first element in your `task-component`.

So just remove `data-task-id="{{ $task->id }}"` from the component call and move it onto your first element in your `task-component.blade.php`

Hope this helps!
